src/core/int128.d

```diff
@@ -0,0 +1,748 @@
+/* 128 bit integer arithmetic.
+ *
+ * Not optimized for speed.
+ *
+ * Copyright: Copyright D Language Foundation 2022.
+ * License:   $(LINK2 http://www.boost.org/LICENSE_1_0.txt, Boost License 1.0)
+ * Authors:   Walter Bright
+ * Source:    $(DRUNTIMESRC core/_int128.d)
+ */
+
+module core.int128;
+
+nothrow:
+@safe:
+@nogc:
+
+alias I = long;
+alias U = ulong;
+enum Ubits = U.sizeof * 8;
+
+struct Cent
+{
+    U lo;      // low 64 bits
+    U hi;      // high 64 bits
+}
+
+enum One = Cent(1);
+enum Zero = Cent();
+enum MinusOne = neg(One);
+
+/*****************************
+ * Test against 0
+ * Params:
+ *      c = Cent to test
+ * Returns:
+ *      true if != 0
+ */
+pure
+bool tst(Cent c)
+{
+    return c.hi || c.lo;
+}
+
+
+/*****************************
+ * Complement
+ * Params:
+ *      c = Cent to complement
+ * Returns:
+ *      complemented value
+ */
+pure
+Cent com(Cent c)
+{
+    c.lo = ~c.lo;
+    c.hi = ~c.hi;
+    return c;
+}
+
+/*****************************
+ * Negate
+ * Params:
+ *      c = Cent to negate
+ * Returns:
+ *      negated value
+ */
+pure
+Cent neg(Cent c)
+{
+    return inc(com(c)); // ~c + 1
```

Comment:
```suggestion
    if (c.lo == 0)
        c.hi = -c.hi;
    else
    {
        c.lo = -c.lo;
        c.hi = ~c.hi;
    }
    return c;
```

Reply:
I like the simplicity of the first for a baseline implementation. There are a lot of optimizations that can and should be done, but they are all harder to understand. Having a baseline gives something to judge it against.